Makefile

```diff
@@ -463,7 +463,7 @@ tidy-check: tidy
 	@diff=$$(git diff --color=always go.mod go.sum $(GO_LICENSE_FILE)); \
 	if [ -n "$$diff" ]; then \
 		echo "Please run 'make tidy' and commit the result:"; \
-		echo "$${diff}"; \
+		printf "%s" "$${diff}"; \
```

Comment:
More places need this treatment?

```
$ rg 'commit the result' Makefile
257:	  echo "Please run 'make fmt' and commit the result:"; \
283:		echo "Please run 'make generate-swagger' and commit the result:"; \
465:		echo "Please run 'make tidy' and commit the result:"; \
881:		echo "Please run 'make svg' and 'git add $(SVG_DEST_DIR)' and commit the result:"; \
892:		echo "Please run 'npm install --package-lock-only' and commit the result:"; \

Reply:
The root problem is that if it uses `echo`, then the `\n` in json file will be outputed as new line. So it needs to use `printf "%s"` to keep the original output.

Other "echo" commands only output simple strings, so no need to use this trick

Reply:
To make code more consistent, I searched `echo "$` and replaced the "diff" ones with "printf" in https://github.com/go-gitea/gitea/pull/33654/commits/89e54b3ca9e950ebe9f384dd0ed501438cdcdf7d